src/admin_ws.rs

```diff
@@ -0,0 +1,187 @@
+use std::{
+    future::Future,
+    sync::{Arc, Mutex},
+};
+
+use holochain_client::{AdminWebsocket, ConductorApiError, ConductorApiResult};
+use tokio::time::{sleep, Duration};
+
+use crate::{HcHttpGatewayError, HcHttpGatewayResult};
+
+const ADMIN_WS_CONNECTION_MAX_RETRIES: usize = 1;
```

Comment:
1 is far too small. That means that as soon as there's been one error, the ws is reconnected and that's it. Another error and the gateway is out of service. I don't think the retries need to be tracked at all. Instead there can be 2-3 reconnection attempts and if they have failed, the gateway could shut down.

Reply:
I've left some similar comments about this retry logic